src/lcm/grids.py

```diff
@@ -30,13 +30,13 @@ class DiscreteGrid(Grid):
 
     """
 
-    options: Collection[int | float]
+    options: type
 
     def __post_init__(self) -> None:
-        if not isinstance(self.options, Collection):
+        if not is_dataclass(self.options):
             raise GridInitializationError(
-                "options must be a collection of scalar int or float values, e.g., a ",
-                "list or tuple",
+                "options must be a dataclass with scalar int or float fields, but is "
```

Comment:
We can be more precise in the description here now. However, I am not sure why we want to distinguish between this error and what is thrown by `_validate_discrete_grid` ?

Reply:
In `_validate_discrete_grid`, I want to assume that I am working with a dataclass. Makes collection of error messages much easier. Be sure to checkout the updated version.